cypress/integration/onlySubscribersCanRate.feature.js

```diff
@@ -0,0 +1,133 @@
+      'https://fakest-newzz.herokuapp.com/api/subscriptions',
+      { paid: 'true', message: 'Thank you for subscribing' }
+    );
+    cy.intercept('POST', 'https://fakest-newzz.herokuapp.com/api/auth', {
+      fixture: 'registration.json',
+    });
+    cy.visit('/');
+  });
+
+  describe('Successfully', () => {
+    beforeEach(() => {
+      cy.intercept('POST', 'https://fakest-newzz.herokuapp.com/api/ratings', {
+        statusCode: 201,
+      });
+    });
+
+    it('is expected to restrict non-subscribers', () => {
+      cy.visit('/articles/4');
+      cy.get('[data-cy=article-container]').within(() => {
+        cy.get('[data-cy=article-rating]').should('contain', '3.7');
+        cy.get('[data-cy=article-rating-button]').find('i').eq(4).click();
+      });
+      cy.get('#rating-message').should(
+        'contain',
+        'You have to subscribe to be able to rate'
+      );
+    });
+
+    it('is expected to allow subscribers to rate', () => {
+      cy.get('[data-cy=login-button]').click();
+      cy.url('/registration');
+      cy.get('[data-cy=login-form]').within(() => {
+        cy.get('[data-cy=registration-button]').click();
+      });
+      cy.get('[data-cy=registration-form]').within(() => {
+        cy.get('[data-cy=registration-first-name]').type('Bob');
+        cy.get('[data-cy=registration-last-name]').type('Kramer');
+        cy.get('[data-cy=registration-email]').type('user@mail.com');
+        cy.get('[data-cy=registration-password]').type('password');
+        cy.get('[data-cy=registration-confirmation-password]').type('password');
+      });
+      cy.wait(1000);
+
+      cy.get('[data-cy=card-details]').within(() => {
+        cy.get('iframe[name^="__privateStripeFrame"]').then(($iframe) => {
+          const $body = $iframe.contents().find('body');
+          cy.wrap($body)
+            .find('input[name="cardnumber"]')
+            .type('4242424242424242', { delay: 50 });
+        });
+      });
+      cy.get('[data-cy=card-details]').within(() => {
+        cy.get('iframe[name^="__privateStripeFrame"]').then(($iframe) => {
+          const $body = $iframe.contents().find('body');
+          cy.wrap($body)
+            .find('input[name="exp-date"]')
+            .type('0424', { delay: 50 });
+        });
+      });
+      cy.get('[data-cy=card-details]').within(() => {
+        cy.get('iframe[name^="__privateStripeFrame"]').then(($iframe) => {
+          const $body = $iframe.contents().find('body');
+          cy.wrap($body).find('input[name="cvc"]').type('424', { delay: 50 });
+        });
+      });
+      cy.get('[data-cy=card-details]').within(() => {
+        cy.get('iframe[name^="__privateStripeFrame"]').then(($iframe) => {
+          const $body = $iframe.contents().find('body');
+          cy.wrap($body)
+            .find('input[name="postal"]')
+            .type('12345', { delay: 50 });
+        });
+      });
+      cy.get('[data-cy=registration-submit]').click();
+      cy.wait(5000);
```

Comment:
thats a long wait, is there a reason?

Reply:
We have a loader that shows after the submit that has a timer of 5 seconds, so we need cypress to wait before continuing with it's tests.